src/redux/init.js

```diff
@@ -0,0 +1,48 @@
+const storedState = JSON.parse(localStorage.getItem("dataState"));
+const todoListsState = storedState.todoLists;
+
+export const getInitialState = () => {
+  Object.keys(todoListsState).forEach(todoListId => {
+    const todoList = todoListsState[todoListId];
+    const todos = todoList.todos;
+
+    Object.keys(todos).forEach(todoId => {
+      const todo = todos[todoId];
+      todo.dueDate = new Date(todo.dueDate);
+    })
+  })
+
+  const initialState = {
+    todoLists: todoListsState,
+    userPreferences: storedState.userPreferences,
+    notifications: { }
+  }
+
+  return initialState
+};
+
+export const afterStoreInitialization = store => {
+  setTimeout(() => triggerNotifications(store), 5000);
+};
+
+const triggerNotifications = store => {
+  setTimeout(() => {
+    Object.keys(todoListsState).forEach(todoListId => {
```

Comment:
Because this method is triggered after 5s, and it could be more, the state that you are reading at this point may be out of sync. In this function you have access to the `store`. 
It would be better to use the current state of the store (using `store.getState()`)  to get the todoList state and trigger the notifications

Reply:
Well done overall on this though :)